diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/WhiteSpaceRule.kt

```diff
@@ -143,40 +164,74 @@ class WhiteSpaceRule : Rule("horizontal-whitespace") {
         }
     }
 
-    private fun handleBinaryOperator(node: ASTNode) {
-        if (node.elementType == COLON && node.treeParent.elementType != TYPE_CONSTRAINT) {
-            return
+    private fun handleColon(node: ASTNode) {
+        when (node.treeParent.elementType) {
+            CLASS, SECONDARY_CONSTRUCTOR, TYPE_CONSTRAINT, TYPE_PARAMETER, OBJECT_DECLARATION -> handleBinaryOperator(node)
+            VALUE_PARAMETER, PROPERTY -> handleToken(node, 0, 1)
+            // fixme: find examples or delete this line
+            else -> log.warn("Colon with treeParent.elementType=${node.treeParent.elementType}, not handled by WhiteSpaceRule")
         }
+    }
 
+    private fun handleBinaryOperator(node: ASTNode) {
         val operatorNode = if (node.elementType == OPERATION_REFERENCE) node.firstChildNode else node
 
         if (node.elementType == OPERATION_REFERENCE && node.treeParent.elementType.let { it == BINARY_EXPRESSION || it == POSTFIX_EXPRESSION } ||
                 node.elementType != OPERATION_REFERENCE) {
-            checkAndFixBinaryOperator(node, operatorNode)
+            val requiredNumSpaces = if (operatorNode.elementType in operatorsWithNoWhitespace) 0 else 1
```

Comment:
basically such logic better to move to a separate methods

Reply:
As discussed, this logic is used only here, so will keep.